### ISSUE TITLE
elm-make permission denied

### ISSUE THREAD

**Issue by pmwhite** [elm-lang/elm-platform#81]
C:\Users\Philip>elm-make
Some new packages are needed. Here is the upgrade plan.

  Install:
    elm-lang/core 2.1.0

Do you approve of this plan? (y/n) y
Downloading elm-lang/core
elm-make: elm-lang-core-5204441: MoveFileEx "elm-lang-core-5204441" "2.1.0": permission denied (Access is denied.)

Fresh install on windows 10.  I tried running as administrator.
I used the http://elm-lang.org/install windows installer.

**Comment by pmwhite** [comment]
I probable should have put this in the elm-package repo.  https://github.com/elm-lang/elm-package/issues/139

**Comment by pmwhite** [comment]
I uninstalled and reinstalled using npm.

I set the ELM_HOME to my user home directory.  Still the same problem.  GHC is the latest version, as is npm.

**Comment by narrative** [comment]
Having the same issue.

    PS C:\Users\erik\Dropbox\elm-test> elm make
    Some new packages are needed. Here is the upgrade plan.
    
      Install:
        elm-lang/core 2.1.0
    
    Do you approve of this plan? (y/n) y
    Downloading elm-lang/core
    elm-make.exe: elm-lang-core-5204441: MoveFileEx "elm-lang-core-5204441" "2.1.0": permission denied (
    Access is denied.)
    PS C:\Users\erik\Dropbox\elm-test>

**Comment by rtfeldman** [comment]
I helped someone fix this tonight. He didn't have Git installed, and when we installed Git (and then manually ran `elm-package install` - not sure if that also helped, but certainly `elm-package` relies on Git being installed), and it worked.

**Comment by pmwhite** [comment]
I just tested it twice and it worked.  I had Git installed before, and I haven't done anything to fix the problem, so I'm not sure why it worked.

**Comment by ilyagr** [comment]
Same problem. The following transcript is from `git bash`, same thing happens from `cmd.exe`. The latest git version & elm version might matter. 

```
[529: 10:43 PM ~/elm-test]$ elm package install
Some new packages are needed. Here is the upgrade plan.

  Install:
    elm-lang/core 3.0.0

Do you approve of this plan? (y/n) y
Downloading elm-lang/core
elm-package.exe: elm-lang-core-243a31b: MoveFileEx "elm-lang-core-243a31b" "3.0.0": permission denied (Access is denied.)
[530: 10:44 PM ~/elm-test]$ git --version
git version 2.6.4.windows.1
[532: 10:44 PM ~/elm-test]$ uname -a
MINGW64_NT-10.0 Blocky 2.4.0(0.292/5/3) 2015-12-12 15:39 x86_64 Msys
[531: 10:44 PM ~/elm-test]$ elm package
elm package 0.16.0
(...)
```

**UPDATE:** I moved the `elm-test` directory to `c:\`, and ran `elm package install` there. It worked. It now works in the original location as well. ????!!!

**Comment by knowingpark** [comment]
I had this issue yesterday. Here's what I noticed.
I was trying to compile an elm file with the reactor in a folder that was in my Dropbox (which is being watched for synchronisation with the network).
When I moved it back to the c:\ drive it all started working
But note: I had been working in the Dropbox folder previous to yesterday without any problems :(

**Comment by tobynet** [comment]
I had same issue today.
**I succeeded to run `elm-make` during disable realtime scanning of McAfee Internet security.**

## Notes:

First, I succeeded to install only once by trying `elm-make`  again and again with enable antivirus.
I guess that the strategy of `elm-make` using rename and antivirus are not a well-matched.

## Environment:

* OS: Windows 10(64bit)
* node: v6.3.1
* npm: 3.10.3
* git: 2.8.1.windows.1
* elm: 0.17.1
* McAfee Internet security: 14.0 R10

**Comment by moneyonrails** [comment]
This is what worked for me on my windows machine.
  1. go to settings on chrome
  2. scroll down and choose "advanced settings"
  3. click on "change proxy settings"
  4. click on the "security tab" 
   5. click on "trusted sites"
  6. click on "Add" button
  7. type "https://github.com/elm-lang/core"
  8. click "Ok" and you should be good to go

**Comment by warren2k** [comment]
Unfortunately, this did not work for me. I was really stoked to go through an Elm tutorial. It sucks that I'm being held up by dev environment setup. I might just forget about Windows and try this on Linux VM instead. Sigh...

**Comment by moneyonrails** [comment]
The above solution that I posted is sporadic. It worked for me at times, and at other times it didn't.
Don't give up on elm for windows like I almost did. The guarantee way to write elm code on windows is to install **cygwin** for windows. So go to cygwin.com and download your version of windows. Add cygwin/bin to path through environment variables and you are good to go. Use the cygwin terminal instead of windows cmd and you are good to go, no more stressing over how to run elm code. Enjoy the wonderful functional language called ELM.

**Comment by warren2k** [comment]
Thanks for the advice, @moneyonrails. I've never used cygwin, but I'll give it a try and see how that goes. I have a couple of laptops, so I may try both Windows and a Linux just to compare and contrast. I appreciate the encouragement.

**Comment by warren2k** [comment]
@moneyonrails I've installed Cygwin; however, I'm not exactly sure how to use it for my purposes. Do I just map my Windows paths to Unix paths, use the Cygwin terminal for all my command line activity, but then still use my Windows text editor (Sublime Text, in my case) as normal?

Is that basically all there is to it?

**Comment by moneyonrails** [comment]
You have to use cygwin as your terminal to run install elm packages and run
your programs. You just have to know how to navigate to the location of
your programs from the cygwin terminal. When you open cygwin terminal, to
go to the C drive of your computer you do
cd ../../cygdrive/c
From there you can go to any directory. Let's say you have the folder where
you want to run elm on your desktop. From the C drive on cygwin you will
then do      cd users/your-username/Desktop
Assuming the name of the project on your desktop is elm-project. You will
then do cd elm-project. All those change directories are done on cygwin.
Now you can install packages. You could use any text editor. I used sublime
text for my elm stuff, install packages on cygwin, and when done I run
elm reactor             on cygwin  and go to localhost:8000 to watch my
code in action.
     To recap you use any text editor to write your elm code. Sublime text
has some elm plugins. If you store your elm project on Documents, and work
with say sublime text, you use cygwin to install packages and run your
code. To go directly to elm-project on Documents folder, you'll do  cd
../../cygdrive/c/users/your-username/Documents/elm-project         and
there you are. Good luck, enjoy
this beautiful functional language.

**Comment by warren2k** [comment]
@moneyonrails So, basically, your answer is yes, LOL. Thanks for the confirmation, and for elaborating on it to ensure I understood. I did also end up creating a Linux VM on a secondary laptop and installing Elm on it, since I like keeping a foot in the Linux world; however, I'd like to be able to write Elm program on the Windows host as well, so I'm going to try using Cygwin on my primary Windows laptop as you've suggested.

Thanks again for your help.

**Comment by jonnydedwards** [comment]
Hi All,
I'm getting this problem with elm-package and elm-repl - tried the Cygwin work around but it didn't work - also tried powersheell,git-bash and bash for Ubuntu. Obviously something kooky with "wonkdow's" file-privileges....

**Comment by moneyonrails** [comment]
It worked for me. Windows is not the best platform for writing software
though. You might want to partition your hard drive and install ubuntu for
easier development.

**Comment by jonnydedwards** [comment]
Ok, so the repl works if it's fired up in a directory that's got all the necessary packages (so "core" in "elm-stuff"?). The packager I just run a few times and it gets there eventually....

**Comment by logicmason** [comment]
McAfee is definitely a problem for the installer.

> It worked for me. Windows is not the best platform for writing software
though. You might want to partition your hard drive and install ubuntu for
easier development.

I don't subscribe to this point of view. A great deal of software development has been done on Windows and continues to be done on Windows. Some ecosystems, such as R, are often a pain on Macs due to libgtk issues, some such as Elm have problems on Windows and others, such as Python, PHP, JS or Java are very robust on both. If wide adoption is the goal, it should run on typical dev environments.

**Comment by evancz** [comment]
It looks like OP's question was resolved. We should close issues after OP gets an answer. Getting help like this is probably better for Slack overall, and if anything needs to be bubbled up, folks who have been around the community longer can open an issue outlining what's going on.
